server/src/main/java/io/druid/guice/http/HttpClientModule.java

```diff
@@ -142,7 +142,8 @@ public HttpClient get()
           .withWorkerCount(config.getNumMaxThreads())
           .withCompressionCodec(
               HttpClientConfig.CompressionCodec.valueOf(StringUtils.toUpperCase(config.getCompressionCodec()))
-          );
+          )
+          .withUnusedConnectionTimeoutDuration(config.getUnusedConnectionTimeout().toStandardDuration());
```

Comment:
can we print a warning log if `unusedConnectionTime >= readTimeout` ?

ideally, we should fail the process startup in that case but that might be backward incompatible and fail process startup for users who might have set their `readTimeout` to smaller values.

Reply:
Thanks @himanshug, added a warning log.